libraries/rush-lib/src/schemas/projects.schema.json

```diff
@@ -0,0 +1,148 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Rush main config File",
+  "description": "The main configuration file for the Rush multi-project build tool. See http://rushjs.io for details.",
+  "type": "object",
+
+  "definitions": {
+    "environmentVariables": {
+      "description": "Environment variables for the package manager",
+      "type": "object",
+      "additionalProperties": {
+        "type": "object",
+        "properties": {
+          "value": {
+            "type": "string"
+          },
+          "override": {
+            "type": "boolean"
+          }
+        },
+        "additionalProperties": false
+      }
+    }
+  },
+
+  "properties": {
+    "$schema": {
+      "description": "Part of the JSON Schema standard, this optional keyword declares the URL of the schema that the file conforms to. Editors may download the schema and use it to perform syntax highlighting.",
+      "type": "string"
+    },
+    "projects": {
```

Comment:
Instead of duplicating this in a second schema, update this and the `rush.json` schema to allow anything in the `projects` property and extract this to its own schema and validate the `projects` object separately.

Reply:
Changed to use a "rush-project-config.schema.json"

Reply:
This schema does not seem to be actually importing or using **rush-project-config.schema.json**.

@Iclanton @D4N14L do we actually have an example of a schema referencing another schema in our code base?  In the past I think **anything.schema.json** was used in this way, but today nothing seems to import it.

My concern is whether VS Code will fetch the URL correctly.

Reply:
From what it looks like, when you do a ref to another json schema, doesn't that schema need to be uploaded to a server first? So I guess I would need to create the rush-project-config.schema.json, upload it, then add the reference in rush.schema.json?